service/notesservice.php

```diff
@@ -112,6 +112,9 @@ public function update ($id, $content, $userId){
 
         // prevent directory traversal
         $title = str_replace(array('/', '\\'), '',  $title);
+        // remove hash and space characters from the beginning of the filename
+        // in case of markdown
+        $title = ltrim($title, ' #');
```

Comment:
ltrim() will trim any character in the character list. Order is not important. "# foo", " #foo", "#foo", " foo", "        foo" will all end up as "foo" after this gets executed.